src/main/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidator.java

```diff
@@ -480,11 +481,79 @@ private void validatePresenceOfSingleYearFilerFields(IntangibleAssetsResource in
             subResourceInvalid = true;
         }
 
+        if (intangibleAssetsResource.getNetBookValueAtEndOfPreviousPeriod() != null) {
+
+            addError(errors, unexpectedData,
+                getJsonPath(intangibleSubResource, NET_BOOK_VALUE_PREVIOUS_PERIOD));
+            subResourceInvalid = true;
+        }
+
+        if (intangibleAssetsResource.getNetBookValueAtEndOfCurrentPeriod() != null) {
+
+            subResourceInvalid = isCostSingleYearSubResourceInvalid(errors, intangibleAssetsResource.getCost(),
+                intangibleSubResource, subResourceInvalid);
+
+            subResourceInvalid = isAmortisationSingleYearSubResourceInvalid(errors,
+                intangibleAssetsResource.getAmortisation(), intangibleSubResource,
+                subResourceInvalid);
+        } else {
+
+            if (hasSingleYearFilerNonNetBookValueFieldsSet(intangibleAssetsResource)) {
+
+                addError(errors, valueRequired, getJsonPath(intangibleSubResource, NET_BOOK_VALUE_CURRENT_PERIOD));
+                subResourceInvalid = true;
+            }
+        }
+
         if(subResourceInvalid) {
             invalidSubResource.add(intangibleSubResource);
         }
     }
 
+    private boolean isCostSingleYearSubResourceInvalid(Errors errors, Cost cost, IntangibleSubResource intangibleSubResource,
+        boolean subResourceInvalid) {
+
+        if(cost == null || cost.getAtPeriodEnd() == null) {
+
+            addError(errors, valueRequired, getJsonPath(intangibleSubResource, COST_AT_PERIOD_END));
+            subResourceInvalid = true;
+        }
+        return subResourceInvalid;
+    }
+
+    private boolean isAmortisationSingleYearSubResourceInvalid(Errors errors, Amortisation amortisation, IntangibleSubResource intangibleSubResource,
+        boolean subResourceInvalid) {
+
+        if (amortisation != null && hasAmortisationFieldsSet(amortisation)
+            && amortisation.getAtPeriodEnd() == null) {
+
+            addError(errors, valueRequired, getJsonPath(intangibleSubResource, AMORTISATION_AT_PERIOD_END));
+            subResourceInvalid = true;
+        }
+        return subResourceInvalid;
+    }
+
+    private void validateCurrentNetBookValue(Errors errors, IntangibleAssetsResource intangibleAssetsResource, IntangibleSubResource intangibleSubResource) {
+
+        Long costAtPeriodEnd = getCostAtPeriodEnd(intangibleAssetsResource);
+
+        Long amortisationAtPeriodEnd = getAmortisationAtPeriodEnd(intangibleAssetsResource);
+
+        Long calculatedCurrentNetBookValue = costAtPeriodEnd - amortisationAtPeriodEnd;
+
+        if(!intangibleAssetsResource.getNetBookValueAtEndOfCurrentPeriod().equals(calculatedCurrentNetBookValue)) {
+
+            addError(errors, incorrectTotal, getJsonPath(intangibleSubResource, NET_BOOK_VALUE_CURRENT_PERIOD));
+        }
+    }
+
+    private Long getNetBookValueAtEndOfCurrentPeriod(IntangibleAssetsResource intangibleAssetsResource) {
```

Comment:
`getNetBookValueAtEndOfCurrentPeriod` is unused.

Reply:
thanks for pointing this out. I have now removed it, as it belongs in the Netbook Total branch.